CONTRIBUTING.md

```diff
@@ -1,33 +1,98 @@
 
-Just use a 'bug report' template when creating a new issue.
+## Submitting issues
 
-## Submit a PR
+Both bug reports and feature requests are welcome.
+Submit issues [here](https://github.com/Kotlin/kotlinx.serialization/issues).
 
-The process of submitting a pull request depends on what kind of changes you have in it:
+* Search for existing issues to avoid reporting duplicates.
+* When submitting a bug report:
+  * Use a 'bug report' template when creating a new issue.
+  * Test it against the most recently released version. It might have been already fixed.
+  * By default, we assume that your problem reproduces in Kotlin/JVM. Please, mention if the problem is
+    specific to Kotlin/JS or Kotlin/Native. 
+  * Include the code that reproduces the problem. Provide the complete reproducer code, yet minimize it as much as possible.
+  * However, don't put off reporting any weird or rarely appearing issues just because you cannot consistently 
+    reproduce them.
+  * If the bug is in behavior, then explain what behavior you've expected and what you've got.  
+* When submitting a feature request:
+  * Use a 'feature request' template when creating a new issue.
+  * Explain why you need the feature &mdash; what's your use-case, what's your domain.
+  * Explaining the problem you face is more important than suggesting a solution. 
+    Report your problem even if you don't have any proposed solution.
+  * If there is an alternative way to do what you need, then show the code of the alternative.
 
-#### Fix typo/documentation clarification
+## Submitting PRs
 
-You can open your PR directly against a `master` branch.
+We love PRs. Submit PRs [here](https://github.com/Kotlin/kotlinx.serialization/pulls).
+However, please keep in mind that maintainers will have to support the resulting code of the project,
+so do familiarize yourself with the following guidelines. 
 
-#### Fix a minor bug
+* All development (both new features and bug fixes) is performed in the `dev` branch.
+  * The `master` branch always contains sources of the most recently released version.
+  * Base PRs against the `dev` branch.
+  * The `dev` branch is pushed to the `master` branch during release.
+  * Documentation in markdown files can be updated directly in the `master` branch, 
+    unless the documentation is in the source code, and the patch changes line numbers.
+* If you fix documentation:
+  * After fixing/changing code examples in the [`docs`](docs) folder or updating any references in the markdown files
+    run the [Knit tool](#running-the-knit-tool) and commit the resulting changes as well. 
+    It will not pass the tests otherwise.
+  * If you plan extensive rewrites/additions to the docs, then please [contact the maintainers](#contacting-maintainers)
+    to coordinate the work in advance.    
+* If you make any code changes:
+  * Follow the [Kotlin Coding Conventions](https://kotlinlang.org/docs/reference/coding-conventions.html). 
```

Comment:
I'd also add that we use star imports (they're not included in default coding conventions)

Reply:
Good catch. Added.